packages/core/template/template.js

```diff
@@ -5,46 +5,50 @@ const fsUtil = require('../src/utils/fsUtil');
 
 const requiredFiles = ['index.md', 'site.json', '_markbind/'];
 
-function Template(rootPath, templatePath) {
-  this.rootPath = rootPath;
-  this.template = templatePath;
-}
+class Template {
+  constructor(rootPath, templatePath) {
+    this.root = rootPath;
+    this.template = templatePath;
+  }
 
-function validateTemplateFromPath(templatePath) {
-  for (let i = 0; i < requiredFiles.length; i += 1) {
-    const requiredFile = requiredFiles[i];
-    const requiredFilePath = path.join(templatePath, requiredFile);
+  static validateTemplateFromPath(templatePath) {
+    for (let i = 0; i < requiredFiles.length; i += 1) {
+      const requiredFile = requiredFiles[i];
+      const requiredFilePath = path.join(templatePath, requiredFile);
 
-    if (!fs.existsSync(requiredFilePath)) {
-      return false;
+      if (!fs.existsSync(requiredFilePath)) {
+        return false;
+      }
     }
+
+    return true;
   }
 
-  return true;
-}
+  static generateSiteWithTemplate(rootPath, templatePath) {
```

Comment:
Should this be static?

Reply:
Same as above - updated according to the suggestion in [ffa06c4](https://github.com/MarkBind/markbind/pull/2148/commits/ffa06c4634c60a65c7aebc0d9adf46b6152df31c)!